src/common/tuklib_cntrl_chars.c

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: 0BSD
+
+///////////////////////////////////////////////////////////////////////////////
+//
+/// \file       tuklib_cntrl_chars.c
+/// \brief      Find and replace single-byte control characters
+//
+//  Author:     Lasse Collin
+//
+///////////////////////////////////////////////////////////////////////////////
+
+#include "tuklib_cntrl_chars.h"
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+
+// strdup() is in SUSv2 and C23. MSVC has it as _strdup().
+#if defined(_MSC_VER) && !defined(strdup)
+#	define strdup _strdup
+#endif
+
+
+extern bool
+tuklib_has_cntrl(const char *str)
+{
+	for (size_t i = 0; str[i] != '\0'; ++i)
+		if (iscntrl((unsigned char)str[i]))
+			return true;
+
+	return false;
+}
+
+
+extern const char *
+tuklib_mask_cntrl(const char *str)
+{
+	static char *mem = NULL;
```

Comment:
Threading is utilized, so this src/common function should probably have static thread_local char *mem to avoid future issues.

Reply:
`thread_local` needs C11 but the codebase is C99 for now. The function is documented to be thread unsafe. It's called from the main thread only.

The function is really primitive as it doesn't even provide more than one memory slot. That is, one cannot print a message that needs two masked strings. It's easy to expand but the first version doesn't need to do more than is required right now.